scotty/pitch_angle_analysis.py

```diff
@@ -0,0 +1,858 @@
+    def plot_cutoff_angles(self, poloidal_angle):
+        return
+
+    def plot_opt_tor_vs_freq(self, poloidal_angle):
+        return
+    """
+
+
+class SweepFromNC(SweepDataset):
+    """Class for reading SweepDataset data written to a .nc file."""
+
+    def __init__(self, path_string):
+        with xr.open_dataset(path_string) as ds:
+            class_type = ds.attrs["class_type"]
+            if class_type != "SweepDataset":
+                raise ValueError(
+                    ".nc file does not have class_type = 'SweepDataset' attribute"
+                )
+            self.dataset = ds
+
+
+class PitchDiagnostic:
+    """Separate class for representing a particular pitch angle diagnostic design/setup.
+    Analysis methods take in Scotty simulation data in a SweepDataset class, and saves the
+    output in a dictionary of datasets. Each dataset corresponds to a type of analysis
+    output with variables corresponding to different SweepDataset equilibriums. Datasets
+    are organized into NetCDF groups when reading and writing to file.
+
+    Note that attributes will be stored as global Dataset attributes rather than as python
+    class attributes for easy writing to a netCDF file.
+
+    Args:
+        Poloidal_Angles (array): Poloidal steering positions for array of microwave antennae
+        Toroidal_Angles (array): Toroidal steering positions for array of microwave antennae
+        Frequencies (array): Discrete frequency coverage of microwave antennaes
+        poloidal_position (array, optional): (R, Z) coordinate launch positions of the antennae,
+        toroidal angle coordinate is not required due to symmetry. Default None assumes that
+        poloidal_position is identical to SweepDataset simulation parameters.
+        curvatures (float/array, optional): Beam curvature for launch beam, if array must be the
+        same length as Frequencies. Default None assumes curvature is identical to SweepDataset
+        simulation parameters.
+    """
+
+    def __init__(
+        self,
+        Poloidal_Angles,
+        Toroidal_Angles,
+        Frequencies,
+        position=None,
+        curvatures=None,
+    ):
+        ds = xr.Dataset()
+        ds.attrs["class_type"] = "PitchAngleDiagnostic"
+        ds.attrs["Poloidal_Angles"] = Poloidal_Angles
+        ds.attrs["Toroidal_Angles"] = Toroidal_Angles
+        ds.attrs["Frequencies"] = Frequencies
+        ds.attrs["position"] = position
+        ds.attrs["curvatures"] = curvatures
+        self.home = ds
+        self.ds_dict = {}
+
+    # get attribute methods
+
+    def get_Poloidal_Angles(self):
+        return self.home.attrs["Poloidal_Angles"]
+
+    def get_Toroidal_Angles(self):
+        return self.home.attrs["Toroidal_Angles"]
+
+    def get_Frequencies(self):
+        return self.home.attrs["Frequencies"]
+
+    def get_position(self):
+        return self.home.attrs["position"]
+
+    def get_curvatures(self):
+        return self.home.attrs["curvatures"]
```

Comment:
Consider making all of these properties with the `@property` decorator (and if so, drop the `get_` prefix from the names)